### PR TITLE
Switch to AutoAPI

### LINKED ISSUE #437: `autosummary` does not list all functions in a module
The main function `ppt_distinguishability` is listed in the documentation but the other two public functions are not listed in the docs. 

https://toqito.readthedocs.io/en/latest/_autosummary/toqito.state_opt.ppt_distinguishability.html#toqito-state-opt-ppt-distinguishability

https://github.com/vprusso/toqito/blob/54bdc08c019b0d00ca963e9ef1aba38e4d616064/toqito/state_opt/ppt_distinguishability.py#L120
https://github.com/vprusso/toqito/blob/54bdc08c019b0d00ca963e9ef1aba38e4d616064/toqito/state_opt/ppt_distinguishability.py#L198

We want to list all the functions in a module. 

Non-local games does appear to do this using `automodule` but it is recommended to use `autosummary`. The issue with `autosummary` is that it does not list all functions in a module.  

https://github.com/vprusso/toqito/blob/master/docs/nonlocal_games.rst#nonlocal-games

https://stackoverflow.com/a/53101990/10241324

Using `recursive` in https://github.com/vprusso/toqito/pull/436/commits/b443ca7b3ac70daee11df83ec970f34d8564e2a1 did not work for `autosummary`. https://stackoverflow.com/a/62613202/10241324

One option is to use `sphinx-apidoc` instead. https://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html#sphinx-apidoc

Another option is to continue using `autosummary` and dig deep into what could work for our purposes. 

To close this issue, make one style consistent across all `rst` files used for `autosummary`/ `automodule`.

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description
Fixes #437 

## Todos
Notable points that this PR has either accomplished or will accomplish.
  -  [ ] TODO 1

## Questions
-  [ ] Question1

## Status
-  [ ] Ready to go

### CODE REVIEW

**Comment by purva-thakre** [comment]
@vprusso What do you think about using an auto tool to create the API reference documentation? Helps us save the work of having different `.rst` files.

**Comment by purva-thakre** [comment]
To do: Fixable warnings.

```
/toqito/docs/autoapi/channel_metrics/fidelity_of_separability/index.rst:8: WARNING: citation not found: Phil23
/toqito/docs/autoapi/nonlocal_games/nonlocal_game/index.rst:37: WARNING: undefined label: '_ref-label-nl-games-tutorial'
/toqito/docs/autoapi/state_metrics/fidelity_of_separability/index.rst:8: WARNING: citation not found: Phil23
```

**Comment by vprusso** [comment]
> @vprusso What do you think about using an auto tool to create the API reference documentation? Helps us save the work of having different `.rst` files.
> 
> The current order of modules in the API reference is a bit jumbled up though. I can work on figuring it out later if you are fine with this change.
> 
> image

I'm certainly on board with automating this process (although I am not sure I'm aware of any specific tool that would be up for this task. Did you have one in mind?

**Comment by purva-thakre** [comment]
> I'm certainly on board with automating this process (although I am not sure I'm aware of any specific tool that would be up for this task. Did you have one in mind?

Ha, yeah. This PR switches to using `AutoAPI`. I wanted some feedback after you take a look at the RTD build. 

image

**Comment by vprusso** [comment]
> Ha, yeah. This PR switches to using `AutoAPI`. I wanted some feedback after you take a look at the RTD build.

Ah gotcha, I just saw and responded to this comment as a starting point. Gotcha. As this is still in draft status, would it make sense for me to review, or should I hold off until it's ready?

**Comment by purva-thakre** [comment]
It's in draft status due to the 2 outstanding comments I made earlier today. 

Just to summarize, this PR 

1. will get rid of most of the `.rst` files used to list the various modules in toqito: `channels.rst`, `matrices.rst`, `measurements.rst`, `nonlocal_games.rst`, `perms.rst`, `rand.rst` and `states.rst`. 
2. will have 1 page called `API reference` which lists all the functions by their respective modules. We don't have to use the files discussed in item 1. I also need to fix the weird ordering of modules in `API Reference`. I think alphabetical order make sense here or somethingelse you might have in mind. 

Are you fine with items 1 and 2?

File: docs/conf.py [hunk 1]
```diff
@@ -49,9 +48,31 @@
...
+    "*/state_ops/tests/*",
+    "*/state_opt/tests/*",
+    "*/channels/tests/*",
+    "*/matrices/tests/*",
+    "*/matrix_ops/tests/*",
+    "*/states/tests/*",
+    "*/channel_props/tests/*",
+    "*/measurement_ops/tests/*",
+    "*/measurement_props/tests/*"]
+autoapi_member_order = 'groupwise'
```
Comment (purva-thakre) [inline docs/conf.py:75]:
```suggestion
autoapi_member_order = 'alphabetical'
```

**Pull request merged**